src/github.com/matrix-org/dendrite/roomserver/storage/sql.go

```diff
@@ -8,25 +8,60 @@ import (
 type statements struct {
 	selectPartitionOffsetsStmt *sql.Stmt
 	upsertPartitionOffsetStmt  *sql.Stmt
+	insertEventTypeNIDStmt     *sql.Stmt
+	selectEventTypeNIDStmt     *sql.Stmt
+	insertEventStateKeyNIDStmt *sql.Stmt
+	selectEventStateKeyNIDStmt *sql.Stmt
+	insertRoomNIDStmt          *sql.Stmt
+	selectRoomNIDStmt          *sql.Stmt
+	insertEventStmt            *sql.Stmt
+	insertEventJSONStmt        *sql.Stmt
 }
 
 func (s *statements) prepare(db *sql.DB) error {
 	var err error
 
-	_, err = db.Exec(partitionOffsetsSchema)
-	if err != nil {
+	if err = s.preparePartitionOffsets(db); err != nil {
 		return err
 	}
 
-	if s.selectPartitionOffsetsStmt, err = db.Prepare(selectPartitionOffsetsSQL); err != nil {
+	if err = s.prepareEventTypes(db); err != nil {
 		return err
 	}
-	if s.upsertPartitionOffsetStmt, err = db.Prepare(upsertPartitionOffsetsSQL); err != nil {
+
```

Comment:
I had to split the `prepare` func up to evade the cyclomatic complexity restrictions, however this approach isn't sustainable. What do you think I should do about it? Should make a list of `prepare` functions somewhere and run them in a loop?